### ISSUE TITLE
No obvious way to find changelogs for previous versions

### ISSUE THREAD

**Issue by wesleytodd** [nodejs/nodejs.org#6627]
### Enter your suggestions in details:

I see I can click the link for "this version" but that is in a modal with no obvious url to visit it directly or link to it. It also means even if I know the version I am looking for I cannot even edit a URL for it. Ideally IMO that link would load a whole new page, not a modal but if not that I think we should make it *really* easy to find **all** changelog links.

**Comment by wesleytodd** [comment]
Looks like the click was intercepted and a cmd-click did open in a new window. That said, I think my "obvious" comment stands as an improvement recommendation.

**Comment by Jay-Karia** [comment]
@wesleytodd 

Node.js full changelog can be found in it's GitHub repo: https://github.com/nodejs/node/tree/main/doc/changelogs

Should we consider making a new page for changelogs which can link to the repo?

**Comment by wesleytodd** [comment]
There used to be a link to the individual change logs for a version from the downloads page. The problem with the current modal window is it doesn't update the browser url, so it is harder to share.

Specifically this link (which now I apparently cannot cmd-click to get a new tab):

<img width="274" alt="Screenshot 2024-08-05 at 10 41 45 AM" src="https://github.com/user-attachments/assets/9e374221-4381-403f-b23f-543a7fdd7bd1">

 Additionally on this, I think that we regressed a few UX things related to finding the metadata for versions. There used to be a single link to view this page: https://nodejs.org/dist/

I used that link all the time, and I dont see anything linking to it anymore. IIRC it used to say something like "other download options" or something?

**Comment by ovflowd** [comment]
> Specifically this link (which now I apparently cannot cmd-click to get a new tab):

Because it is not a link, just a modal. Ultimately, the blog post should suffice if you want a shareable link.

> Additionally on this, I think that we regressed a few UX things related to finding the metadata for versions. There used to be a single link to view this page: [nodejs.org/dist](https://nodejs.org/dist/)

For the end user, there's no point seeing nodejs.org/dist; Very few users would know what is what and what to download. That's why each tab has specific download methods for what they might be searching for; It's like throwing them at a huge Netflix catalog (pun not intentionally intended) versus showing them "series we recommend for you"

> I used that link all the time, and I dont see anything linking to it anymore. IIRC it used to say something like "other download options" or something?

The closest to this, which remains true is

![image](https://github.com/user-attachments/assets/b884a1d4-1a8e-49b8-9800-a6d42bb7e451)

On the "Prebuilt binaries" tab, as they are all Prebuilt binaries. I'm definitely open for suggestions :)

**Comment by wesleytodd** [comment]
> Because it is not a link, just a modal. Ultimately, the blog post should suffice if you want a shareable link.

No I dont want a blog post though lol. I want the changelog. I use this to find and debug issues, not just learn about the release. I often find the specific thing I think is broken and run a node version right before it to check. I really would like to make the case for bringing back links to a full page change log view for the release, even if that just links to GH.

---

Sorry early send.

> For the end user, there's no point seeing nodejs.org/dist; Very few users would know what is what and what to download. 

Yeah but I use it. I use it for all sorts of things, like checking on my phone which versions of things ship with what (I do support for folks, and sometimes and doing it from my phone). I agree most users dont need it, but for those of us that do I think a small link makes sense.

**Comment by ovflowd** [comment]
> No I dont want a blog post though lol. I want the changelog. I use this to find and debug issues, not just learn about the release. I often find the specific thing I think is broken and run a node version right before it to check. I really would like to make the case for bringing back links to a full page change log view for the release, even if that just links to GH.

You are aware the changelog is embedded on the blog post, right? It is even more complete there, in general.

**Comment by wesleytodd** [comment]
Sure, but I dont want to read or find it in there. I want the plain old changelog which still exists and was always available easily from that link. Ideally I have the running changlog so I can search through all releases for keywords. Like this is the link I need and I usually dont want to go digging in the GH repo for it. https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md

**Comment by ovflowd** [comment]
> Yeah but I use it. I use it for all sorts of things, like checking on my phone which versions of things ship with what (I do support for folks, and sometimes and doing it from my phone). I agree most users dont need it, but for those of us that do I think a small link makes sense.

Then just go nodejs.org/dist -- why should this link be listed on the website? 😅 again, your use case is very specific.

**Comment by wesleytodd** [comment]
> your use case is very specific.

I dont see this being only a me thing. But I am not going to die on this hill.

**Comment by ovflowd** [comment]
> Sure, but I dont want to read or find it in there. I want the plain old changelog which still exists and was always available easily from that link. Ideally I have the running changlog so I can search through all releases for keywords. Like this is the link I need and I usually dont want to go digging in the GH repo for it. [nodejs/node@`main`/doc/changelogs/CHANGELOG_V22.md](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V22.md?rgh-link-date=2024-09-10T13%3A48%3A32Z)

We never had a changelog button. https://web.archive.org/web/20180308115257/https://nodejs.org/en/download/

**Comment by ovflowd** [comment]
> > your use case is very specific.
> 
> I dont see this being only a me thing. But I am not going to die on this hill.

Can you prove to me this is something other people used? Our designs are backed by data on what people clicked or not. We recorded thousands of user sessions via Sentry and still record and actively monitor engagement and people's actions.

I'm not saying I triple-checked that now, but that back then, we made these decisions conscious with said data. I can double-check if it was actively used, but I doubt.

**Comment by ovflowd** [comment]
Again, it's not something to die on a hill for, but just the more links and the more things we add to the website, the more confusing it can be for an end-user. Our website is targeted at the absolute rookies who have no idea what they are doing -- and needs to be tailored so that it is safe for them to go around, and that gives them confidence.

A link to "All download options" that goes to a default "no index" directory listing... It is the opposite of that.

Regarding the changelog button, it was used somewhat, but only on the homepage. What I can do here for you, @wesleytodd, is add a link to changelog on GitHub within the modal that we open, or we can use regular `a' elements, but by default clicking on it opens the modal (`onClick`) and then all other commands should still work, like open on a new tab, copy link, etc. Would that work for you? And if yes, which option is desirable? A link on the modal or the modal itself being a link?

**Comment by wesleytodd** [comment]
> Regarding the changelog button, it was used somewhat, but only on the homepage. What I can do here for you, @wesleytodd, is add a link to changelog on GitHub within the modal that we open

This sounds like a great place to put it. I agree and totally understand your data driven approach, and don't mean my report here to be enough to counter that decision.  I honeslty thought maybe others were wanting the same thing so figured it was better open the issue. If that is not the case, fine, but I do think this proposal you made is a good middle ground.

**Comment by ovflowd** [comment]
No worries, again, I could be wrong as these decisions were done a time ago.

Ill proceed with the middle ground approach for the changelog.

But I won't be adding a button to go to nodejs.org/dist

**Comment by bmuenzenmeyer** [comment]
@wesleytodd As part of fixing #7060 I wondered if https://nodejs.org/en/about/previous-releases#looking-for-latest-release-of-a-version-branch suffices.

It's such a simple question I thought I'd ask before we go too much further down another path.

**Comment by J-dev740** [comment]
Hi @wesleytodd  I think I can work on this issue, can you assign it to me

**Comment by wesleytodd** [comment]
@bmuenzenmeyer, that is a great improvement. I still think it would be great to just make that modal go away entirely because it is not linkable. It is generally BAD UX to not have something you can link to that is as important as a change log. I would love if someone like @J-dev740 took it on and could make it a normal page with a url like any other.

**Comment by ovflowd** [comment]
I think this got implemented already, so closing it for the meantime!
